src/NKSolver/NKSolvers.F90

```diff
@@ -3043,15 +3061,15 @@ subroutine physicalityCheckANK(lambdaP)
     real(kind=realType), pointer :: wvec_pointer(:)
     real(kind=realType), pointer :: dvec_pointer(:)
     real(kind=alwaysRealType) :: lambdaL ! L is for local
-    real(kind=realType) :: ratio
-
+    real(kind=alwaysRealType) :: tmp ! to receive the global step
```

Comment:
I would prefer a name like `lambdaP_tmp` so it's easier to keep track of what's happening.

Reply:
changed all instances of tmp in the two physicality checks to `lambdaP_recv`

Reply:
This works for me